python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,464 @@
+            index positions. The Series contains values from all columns of the
+            DataFrame.
+        """
+        if isinstance(self.stats, pd.Series):
+            self.logger.info("Stats are already packed into a Series.")
+            return self.stats
+
+        self.logger.info("Packing stats into a Series.")
+        df = self.stats.transpose()
+
+        # Define the prefix patterns
+        column_prefixes = df.columns
+        index_positions = df.index
+
+        # Generate all combinations of prefixes and positions
+        index_prefixes = [
+            f"measuredForce{stat.capitalize()}{position}"
+            for stat in index_positions
+            for position, _ in enumerate(column_prefixes)
+        ]
+
+        # Flatten the DataFrame and set the new index
+        result_series = df.stack().reset_index(drop=True)
+        result_series.index = index_prefixes
+
+        # Append the event information to the Series
+        event_dict = vars(self.event)
+        event_dict = {
+            key: val
+            for key, val in event_dict.items()
+            if key in ["dayObs", "seqNum", "version"]
+        }
+
+        # Create a pandas Series from the dictionary
+        event_series = pd.Series(event_dict)
+
+        # Concatenate the two Series
+        result_series = pd.concat([event_series, result_series])
+
+        # Display the resulting Series
+        return result_series
+
+
+def find_adjacent_true_regions(
+    series: pd.Series, min_adjacent: None | int = None
+) -> list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]:
+    """
+    Find regions in a boolean Series containing adjacent True values.
+
+    Parameters
+    ----------
+    series : pd.Series
+        The boolean Series to search for regions.
+
+    min_adjacent : int, optional
+        Minimum number of adjacent True values in a region.
+        Defaults to half size of the series.
+
+    Returns
+    -------
+    list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]
+        A list of tuples representing the start and end indices of regions
+        containing more than or equal to min_adjacent adjacent True values.
+    """
+    min_adjacent = min_adjacent if min_adjacent else 0.5 * series.size
+    regions = []
+    for key, group in series.groupby((series != series.shift()).cumsum()):
+        if key and len(group) >= min_adjacent:
+            region_indices = group.index
+            regions.append((region_indices.min(), region_indices.max()))
+    return regions
+
+
+def get_tma_slew_event(day_obs: int, seq_number: int) -> TMAEvent:
+    """
+    Retrieve Telescope Mount Assembly (TMA) slew events within a specified time
+    range.
+
+    Parameters
+    ----------
+    dayObs : int
+        Observation day in the YYYYMMDD format.
+    seqNum : int
+        Sequence number associated with the slew event.
+
+    Returns
+    -------
+    lsst.summit.utils.tmaUtils.TMAEvent
```

Comment:
I think this needs a variable name, so something like:
`event : lsst.summit.utils.tmaUtils.TMAEvent`